cmake/modules/FindTAEF.cmake

```diff
@@ -55,16 +66,11 @@ else()
   message(ERROR "Unable to find TAEF binaries.")
 endif()
 
-if ((CMAKE_GENERATOR_PLATFORM STREQUAL "x64") OR ("${CMAKE_C_COMPILER_ARCHITECTURE_ID}" STREQUAL "x64"))
-  set(TAEF_ARCH "x64")
-elseif ((CMAKE_GENERATOR_PLATFORM STREQUAL "x86") OR ("${CMAKE_C_COMPILER_ARCHITECTURE_ID}" STREQUAL "x86"))
-  set(TAEF_ARCH "x86")
-elseif ((CMAKE_GENERATOR_PLATFORM MATCHES "ARM64.*") OR ("${CMAKE_C_COMPILER_ARCHITECTURE_ID}" MATCHES "ARM64.*"))
-  set(TAEF_ARCH "arm64")
-elseif ((CMAKE_GENERATOR_PLATFORM MATCHES "ARM.*") OR ("${CMAKE_C_COMPILER_ARCHITECTURE_ID}" MATCHES "ARM.*"))
-  set(TAEF_ARCH "arm")
-endif((CMAKE_GENERATOR_PLATFORM STREQUAL "x64") OR ("${CMAKE_C_COMPILER_ARCHITECTURE_ID}" STREQUAL "x64"))
-
+# taef/build/Binaries use amd64 for x64.
+# For hlsl test, use x64 for x64.
+if (TAEF_ARCH STREQUAL "amd64")
+    set(TAEF_ARCH "x64")
```

Comment:
I don't like the redefinition of the same variable, that's scary.  Can we use two separate variables if they have to be different?  `TAEF_TEST_ARCH` perhaps?

Reply:
Yea... this is odd. Why not have the if/elseif blob above set `TAEF_ARCH` and `TAEF_EXTERNAL_PATH` as separate variables, then you only have one conditional block.

Reply:
Will remove the redef.
It will be useless after https://github.com/microsoft/DirectXShaderCompiler/pull/4973 remove the filter for filter conflict with name for taef test.

Reply:
Removed.